AttributeError with asyncpg v0.29.0

A new release of asyncpg has just been published. It looks like the attribute `asyncpg.Pool._working_addr` does not exist any more (https://github.com/MagicStack/asyncpg/commit/4ddb039746a99b31cc4724329dd600b87a3f9469). But this attribute is expected in the method `hasql.PoolManager.host` (https://github.com/aiokitchen/hasql/blob/ecbf8576a0aa9f8e958b53b5a4f0a6f1736bf54a/hasql/asyncpg.py#L45C1-L47). This results in an `AttributeError` when using the latest version of asyncpg.

```
    def host(self, pool: asyncpg.Pool):
>       addr, _ = pool._working_addr
E       AttributeError: 'Pool' object has no attribute '_working_addr'
```

--- a/hasql/asyncpg.py
+++ b/hasql/asyncpg.py
@@ -1,7 +1,8 @@
 import asyncio
-from typing import Sequence
+from typing import ClassVar, Dict, Sequence
 
 import asyncpg  # type: ignore
+from packaging.version import parse as parse_version
 
 from hasql.base import BasePoolManager
 from hasql.metrics import DriverMetrics
@@ -10,6 +11,7 @@ from hasql.utils import Dsn
 
 class PoolManager(BasePoolManager):
     pools: Sequence[asyncpg.Pool]
+    cached_hosts: ClassVar[Dict[int, str]] = {}
 
     def get_pool_freesize(self, pool):
         return pool._queue.qsize()
@@ -42,9 +44,23 @@ class PoolManager(BasePoolManager):
     def is_connection_closed(self, connection):
         return connection.is_closed()
 
-    def host(self, pool: asyncpg.Pool):
-        addr, _ = pool._working_addr
-        return addr
+    if parse_version(asyncpg.__version__) >= parse_version("0.29.0"):
+        # We try to reproduce the same behaviour of the _working_addr
+        # attribute prior the 0.29.0 version for getting the host
+        # linked to a pool.
+        def host(self, pool: asyncpg.Pool):
+            conn = next(
+                (holder._con for holder in pool._holders if holder._con),
+                None
+            )
+            if conn is not None:
+                addr, _ = conn._addr
+                PoolManager.cached_hosts[id(pool)] = addr
+            return PoolManager.cached_hosts[id(pool)]
+    else:
+        def host(self, pool: asyncpg.Pool):
+            addr, _ = pool._working_addr
+            return addr
 
     def _driver_metrics(self) -> Sequence[DriverMetrics]:
         return [

--- a/setup.py
+++ b/setup.py
@@ -58,7 +58,7 @@ setup(
             "async_timeout",
             "psycopg[pool]>=3.0,<4",
             "aiopg[sa]~=1.4.0",
-            "asyncpg~=0.27.0",
+            "asyncpg~=0.29.0",
             "pytest~=6.2.5",
             "pytest-cov~=3.0.0",
             "aiomisc~=15.2.4",

